GitUI/UserControls/FileStatusList.cs

```diff
@@ -295,38 +297,18 @@ public override ContextMenuStrip ContextMenuStrip
 
         public bool FilterFocused => FilterComboBox.Focused;
 
-        public bool FilterVisible
+        private void SetFileStatusListVisibility(bool filesPresent)
         {
-            get { return _filterVisible; }
-            set
-            {
-                if (value == _filterVisible)
-                {
-                    return;
-                }
-
-                _filterVisible = value;
-                FilterVisibleInternal = value;
-            }
-        }
-
-        private bool FilterVisibleInternal
-        {
-            get => FilterComboBox.Visible;
-            set
-            {
-                FilterComboBox.Visible = value;
-                SetDeleteFilterButtonVisibility();
-                SetFilterWatermarkLabelVisibility();
-
-                int top = value
-                    ? FileStatusListView.Margin.Top + FilterComboBox.Bottom + FilterComboBox.Margin.Bottom
-                    : FileStatusListView.Margin.Top;
+            LoadingFiles.Visible = false;
+            NoFiles.Visible = !filesPresent;
+            FilterComboBox.Visible = filesPresent;
 
-                int height = ClientRectangle.Height - FileStatusListView.Margin.Bottom - top;
+            SetDeleteFilterButtonVisibility();
+            SetFilterWatermarkLabelVisibility();
 
-                FileStatusListView.SetBounds(0, top, 0, height, BoundsSpecified.Y | BoundsSpecified.Height);
-            }
+            int top = FilterComboBox.Visible ? FilterComboBox.Bottom + FilterComboBox.Margin.Bottom : 0;
+            int height = ClientRectangle.Height - top - FileStatusListView.Margin.Top - FileStatusListView.Margin.Bottom;
```

Comment:
```suggestion
            int top = FilterComboBox.Visible ? FilterComboBox.Bottom + FilterComboBox.Margin.Bottom + FileStatusListView.Margin.Top : 0;
            int height = ClientRectangle.Height - top;
```

> Note
> Setting the [Margin](https://learn.microsoft.com/en-us/dotnet/api/system.windows.forms.control.margin?view=windowsdesktop-8.0) property on a docked control has no effect on the distance of the control from the edges of its container.

Anyway, shouldn't we let the parent control do the job - by means of `DockStyle.Top` and `DockStyle.Fill`?
Otherwise, we would need to manually adjust on resize, too.

Reply:
I have spent the last weeks trying to get working... 
It could  be working in this setup I believe, but with the SearchBox that may be disabled the order of the boxes can be switched, I did not find a way to fix that.

Reply:
`+ FileStatusListView.Margin.Top` is missing yet - although the value is 0.

Reply:
I had ignored that to limit the line length, now added